x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -704,6 +704,12 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                     .build(),
                 // APM telemetry queries APM indices in kibana task runner
                 RoleDescriptor.IndicesPrivileges.builder().indices("apm-*").privileges("read", "read_cross_cluster").build(),
+                RoleDescriptor.IndicesPrivileges.builder().indices("logs-apm.*").privileges("read", "read_cross_cluster").build(),
+                RoleDescriptor.IndicesPrivileges.builder().indices("logs-apm-*").privileges("read", "read_cross_cluster").build(),
+                RoleDescriptor.IndicesPrivileges.builder().indices("metrics-apm.*").privileges("read", "read_cross_cluster").build(),
+                RoleDescriptor.IndicesPrivileges.builder().indices("metrics-apm-*").privileges("read", "read_cross_cluster").build(),
```

Comment:
Given our datastreams [docs](https://www.elastic.co/guide/en/apm/guide/8.1/apm-data-streams.html) and [code](https://github.com/elastic/apm-server/tree/main/apmpackage/apm/data_stream)
```
Application traces: traces-apm-<namespace>
RUM and iOS agent application traces: traces-apm.rum-<namespace>
APM internal metrics: metrics-apm.internal-<namespace>
APM profiling metrics: metrics-apm.profiling-<namespace>
Application metrics: metrics-apm.app.<service.name>-<namespace>
logs-apm.error-<namespace>
```

```diff
apm-*
logs-apm.*
logs-apm-*"
metrics-apm.*
-metrics-apm-*
traces-apm.*
traces-apm-*
```

I don't think we strictly need access to the `apm` dataset name under the `metrics` type (`{type}-{dataset}-{namespace}`)

Reply:
@Mpdreamz does the same apply to `logs-apm-*`? I don't see that in your list either.